baybe/parameters/substance.py

```diff
@@ -149,6 +153,10 @@ def comp_df(self) -> pd.DataFrame:  # noqa: D102
             else:
                 comp_df = df_uncorrelated_features(comp_df, threshold=self.decorrelate)
 
+        # Add noise to degenerate rows, if present
+        if comp_df.duplicated().any():
```

Comment:
Not necessary requesting you to change, just wanted to raise awareness: in case there are duplicates, the duplicate search is executed twice (once here, once within the method). While probably not a big performance issue, one alternative would be to simply *always* run the method, do the check in there as a guard clause, and exit early if there is nothing to do.

Reply:
ok this already works but I guess the function should have the check and exit prematurely in that case?

Reply:
changed